Backport from 0.9.7b -- don't set the engine for all methods -- it may not
support all of them.
diff -uNr openssl-0.9.7a/crypto/engine/eng_fat.c openssl-0.9.7b/crypto/engine/eng_fat.c
--- openssl-0.9.7a/crypto/engine/eng_fat.c	2002-06-13 13:38:54.000000000 -0400
+++ openssl-0.9.7b/crypto/engine/eng_fat.c	2003-03-13 15:23:19.000000000 -0500
@@ -66,18 +66,18 @@
 	if((flags & ENGINE_METHOD_DIGESTS) && !ENGINE_set_default_digests(e))
 		return 0;
 #ifndef OPENSSL_NO_RSA
-	if((flags & ENGINE_METHOD_RSA) & !ENGINE_set_default_RSA(e))
+	if((flags & ENGINE_METHOD_RSA) && !ENGINE_set_default_RSA(e))
 		return 0;
 #endif
 #ifndef OPENSSL_NO_DSA
-	if((flags & ENGINE_METHOD_DSA) & !ENGINE_set_default_DSA(e))
+	if((flags & ENGINE_METHOD_DSA) && !ENGINE_set_default_DSA(e))
 		return 0;
 #endif
 #ifndef OPENSSL_NO_DH
-	if((flags & ENGINE_METHOD_DH) & !ENGINE_set_default_DH(e))
+	if((flags & ENGINE_METHOD_DH) && !ENGINE_set_default_DH(e))
 		return 0;
 #endif
-	if((flags & ENGINE_METHOD_RAND) & !ENGINE_set_default_RAND(e))
+	if((flags & ENGINE_METHOD_RAND) && !ENGINE_set_default_RAND(e))
 		return 0;
 	return 1;
 	}
